File: Adapter/python/Adapter.py
import abc
from abc import ABCMeta


class IScale(metaclass=ABCMeta):

    @abc.abstractmethod
    def get_weight(self) -> float:
        pass


class RussianScales(IScale):

    def __init__(self, cw: float):
        self.__current_weight = cw

    def get_weight(self) -> float:
        return self.__current_weight


class BritishScales:

    def __init__(self, cw: float):
        self.__current_weight = cw

    def get_weight(self) -> float:
        return self.__current_weight


class AdapterForBritishScales(IScale):

    def __init__(self, british_scales: BritishScales):
        self.__british_scales = british_scales

    def get_weight(self) -> float:
        return self.__british_scales.get_weight() * .453


if __name__ == '__main__':
    kg: float = 55
    lb: float = 55

    rScales = RussianScales(kg)
    bScales = AdapterForBritishScales(BritishScales(lb))

    print(rScales.get_weight())
    print(bScales.get_weight())

File: Adapter/c++/Adapter.cpp
#include <iostream>

class Scales
{
    public:
        virtual float GetWeight() = 0;
};

class RussianScales : public Scales
{
    private:
        float currentWeight;

    public:
        RussianScales(float cw) : currentWeight(cw) {}
        float GetWeight() override;
};

class BritishScales
{
    private:
        float currentWeight;

    public:
        BritishScales(float cw) : currentWeight(cw) {}
        float GetWeight();
};

class AdapterForBritishScales : public Scales
{
    private:
        BritishScales* britishScales;

    public:
        AdapterForBritishScales(BritishScales* bs) : britishScales(bs) {}
        ~AdapterForBritishScales();
        float GetWeight() override;
};

int main()
{
    float kg = 55.0;
    float lb = 55.0;

    Scales* rScales = new RussianScales(kg);
    Scales* bScales = new AdapterForBritishScales(new BritishScales(lb));

    std::cout << rScales->GetWeight() << std::endl;
    std::cout << bScales->GetWeight() << std::endl;
    
    return 0;
}

float RussianScales::GetWeight()
{
    return currentWeight;
}

float BritishScales::GetWeight()
{
    return currentWeight;
}

AdapterForBritishScales::~AdapterForBritishScales()
{
    delete britishScales;
}

float AdapterForBritishScales::GetWeight()
{
    return britishScales->GetWeight() * 0.453;
}